### PR TITLE
Simplify 404/500 page

### PR DESCRIPTION
**Pull request by wxiaoguang**
It doesn't need to waste user's bandwidth to download a 404.png/500.png when error occurs.

ps: I think we do not need to translate "Not Found" and "Internal Server Error", they are widely used internet terms IMO, and they are clearer than the previous images (which only show a number)

<details>

image

image

image

image

</details>

### CODE REVIEW

**Comment by wxiaoguang** [comment]
> Maybe put on two lines?
> 
> ```
>        404 (48px)
>     Not Found (32px)
> ```

No idea which looks really better. Fee free to edit it.

Personally I don't think it's worth to make it that complex, most users do not use these pages daily.

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: web_src/css/base.css [hunk 1]
```diff
@@ -477,6 +477,15 @@ img.ui.avatar,
   padding-bottom: 80px;
 }
 
+.status-page-error-code {
+  font-size: 3em;
+  line-height: initial;
+  text-align: center;
+  padding: 2em 0;
+  font-weight: var(--font-weight-bold);
+  color: var(--color-text-light-3);
```
Comment (silverwind) [inline web_src/css/base.css]:
```suggestion
  color: var(--color-text-light-1);
```

No reason to have so little contrast.

Comment (wxiaoguang) [reply web_src/css/base.css]: I have tried, I think `--color-text-light-3` looks better with large fonts.

Feel free to checkout and edit this PR. Maybe we could make the fonts smaller to 2em or 32px.

Comment (silverwind) [reply web_src/css/base.css]: Hmm yes let me actually try it. Maybe value `2` is also good. `3` seems maybe a bit too subtle.

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Some tweaks done. Ideally I'd want to vertically center on screen but that's actually not so cleanly possible with the current CSS so I left it unchanged.

Screenshot 2024-06-20 at 01 44 33

Screenshot 2024-06-20 at 01 44 24

Screenshot 2024-06-20 at 01 44 43

File: web_src/css/base.css (hunk 1 above)
Comment (silverwind) [reply web_src/css/base.css]: 3 looks ok actually to preverse a bit distinction between title and detail message.

Edit: Ah I guess 2 is better, changed it 😆

**Comment by silverwind** [comment]
Actually changed the color to `--color-text-light-2`:

image

**Comment by wxiaoguang** [comment]
Wrong position.

image

**Comment by silverwind** [comment]
Fixed, the asymmetry is ugly, but I guess that text above would need to be separated out of the `<pre>` block to fix that as well.

image

**Comment by wxiaoguang** [comment]
image

image

image

**Comment by silverwind** [comment]
image

Maybe reduce spacing below code a small bit?

**Comment by silverwind** [comment]
Tweaked once more. I think it looks more visually pleasing if vertical margin between elements is roughly the same.

Screenshot 2024-06-20 at 23 48 56
Screenshot 2024-06-20 at 23 48 44

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Hold on with merge, I will try to vertically center on screen.

**Comment by silverwind** [comment]
Vertical centering added. Took a bit more CSS than I would have liked but the page layout is quite complex unfortunately.

Screenshot 2024-06-21 at 00 14 03
Screenshot 2024-06-21 at 00 14 27

**Comment by silverwind** [comment]
Let's wait for @wxiaoguang to have a final look.

**Comment by wxiaoguang** [comment]
> Vertical centering added. Took a bit more CSS than I would have liked but the page layout is quite complex unfortunately.

I really dislike using too many CSS tricks for such a trivial page.

**Comment by wxiaoguang** [comment]
And it is even totally wrong! I will revert all unnecessary changes and won't spend time on such a trivial PR.

<details>

image

</details>

**Comment by silverwind** [comment]
> I really dislike using too many CSS tricks for such a trivial page.

As I said, the complex CSS is because the page layout is bad and could still drastically simplified. 

> And it is even totally wrong! I will revert all unnecessary changes and won't spend time on such a trivial PR.

I will iterate on it and vertically center it. I think it's important to have well-designed pages.

**Comment by wxiaoguang** [comment]
> > I really dislike using too many CSS tricks for such a trivial page.
> 
> As I said, the complex CSS is because the page layout is bad and could still drastically simplified.

Hacking tricks don't really work well. There are different cases:

* 404 full page
* 404 embedded page with repo header
* 500 full page
* 500 in a partially rendered page, the error message part could appear at any place, unpredictable.

> > And it is even totally wrong! I will revert all unnecessary changes and won't spend time on such a trivial PR.
> 
> Would have been a trivial fix. I will iterate on it and vertically center it. I think it's important to have well-designed pages and especially on big screen it looks bad with the fixed margin currently.

I believe in: correctness > good maintainability > feature/enhancement > base maintainability > unclear patching for blockers > tricky fine-tunes.

For most UI alignment problems, we have fixed them with "base maintainability". But for this one, it only seems to be "tricky fine-tunes" at the moment.

**Comment by wxiaoguang** [comment]
If you would really to fine tune, try this: cb1462d68fdb052e3ebd5018562d5af12a087173


<details>

image


image

</details>

**Comment by wxiaoguang** [comment]
Anything else to change? If no, leave more fune-tunes to the future.

At least, this change is already better than before and doesn't introduce new problems (the UI didn't align in old code either)

**Comment by silverwind** [comment]
The centering looks slightly off, maybe I will tune later.

**Pull request merged**